nixos/modules/services/networking/opsdroid.nix

```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (lib) mkEnableOption mkPackageOptionMD mkOption;
+  cfg = config.services.opsdroid;
+  yaml = pkgs.formats.yaml { };
+  settingsFile = yaml.generate "opsdroid.yaml" cfg.settings;
+in
+{
+  meta.maintainers = [ lib.maintainers.raitobezarius ];
+  options.services.opsdroid = {
+    enable = mkEnableOption "opsdroid, bot framework";
+    package = mkPackageOptionMD pkgs "opsdroid" { };
+    secretsFile = mkOption {
```

Comment:
Just making sure - there's no usecase for this without secrets? Else, could we default this to `null`, to avoid forcing the user to set it?